src/gui/plugins/resource_spawner/ResourceSpawner.cc

```diff
@@ -42,13 +47,36 @@ namespace ignition::gazebo
     public: transport::Node node;
 
     /// \brief The grid model that the qml gridview reflects
-    public: GridModel gridModel;
+    public: ResourceModel resourceModel;
 
-    /// \brief The path list model that the qml treeview reflects
+    /// \brief The path list model that the qml treeview reflects for local
+    /// resources
     public: PathModel pathModel;
+
+    /// \brief The owner list model that the qml treeview reflects for fuel
+    /// resources
+    public: PathModel ownerModel;
+
+    /// \brief Client used to download resources from Ignition Fuel.
+    public: std::unique_ptr<ignition::fuel_tools::FuelClient>
+            fuelClient = nullptr;
+
+    /// \brief Stores details about all Fuel servers providing assets.
+    /// The key is the server URI and the value is the list of models
+    /// corresponding to that URI.
+    public: std::unordered_map<std::string,
+            std::vector<ignition::fuel_tools::ModelIdentifier>> fuelDetails;
+
+    /// \brief The map to cache resources after a search is made on an owner,
+    /// reduces redundant searches
+    public: std::unordered_map<std::string,
+            std::vector<Resource>> ownerModelMap;
   };
 }
 
+static int gridIndex = 0;
+static int localGridIndex = 0;
```

Comment:
These static variables could interfere different instances of the plugin. How about making them members of `ResourceModel`? You could access them from `ResourceSpawner` through `this->resourceModel`.

Reply:
Good catch, fixed in fdff6c8149c682c4141e5110e3b31d8618730867